flink-state-backends/flink-statebackend-forst/src/main/java/org/apache/flink/state/forst/ForStStateRequestClassifier.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.state.forst;
+
+import org.apache.flink.runtime.asyncprocessing.StateRequest;
+import org.apache.flink.runtime.asyncprocessing.StateRequestContainer;
+import org.apache.flink.runtime.asyncprocessing.StateRequestType;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * The ForSt {@link StateRequestContainer} which can classify the state requests by ForStDB
+ * requestType (Get、Put or Iterator).
+ */
+public class ForStStateRequestClassifier implements StateRequestContainer {
+
+    private final List<ForStDBGetRequest<?, ?>> dbGetRequests;
+
+    private final List<ForStDBPutRequest<?, ?>> dbPutRequests;
+
+    public ForStStateRequestClassifier() {
+        this.dbGetRequests = new ArrayList<>();
+        this.dbPutRequests = new ArrayList<>();
+    }
+
+    @Override
+    public void offer(StateRequest<?, ?, ?> stateRequest) {
+        if (stateRequest.getState() == null) {
```

Comment:
when does `stateRequest.getState() == null`?

Reply:
The state could be null only for {@link StateRequestType#SYNC_POINT}. The ForStateBackend does nothing for {@link StateRequestType#SYNC_POINT}. 

I have added some comments about this if-branch.

Reply:
`StateRequestType#SYNC_POINT` would be added to the active buffer, it might be better to add a check here.

https://github.com/apache/flink/blob/master/flink-runtime/src/main/java/org/apache/flink/runtime/asyncprocessing/StateRequestBuffer.java#L125-L127

Reply:
ah... The `SYNC_POINT` will not be handled by the executor and instead, it is completed once it reach the active buffer.

Reply:
Thanks for the reminding. I have removed this if-branch.